mysql-test/suite/innodb/t/parallel_dblwr_encrypt.test

```diff
@@ -0,0 +1,92 @@
+# Test Parallel double write buffer encryption
+
+--source include/have_innodb.inc
+--source include/not_embedded.inc
+--source include/have_debug.inc
+
+SET GLOBAL innodb_page_cleaner_disabled_debug=ON;
+--let $MYSQLD_DATADIR=`SELECT @@datadir`
+
+CREATE TABLE t1(a TEXT) ENCRYPTION='Y';
+
+INSERT INTO t1 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t1 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t1 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t1 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t1 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t1 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+
+CREATE TABLE t2(a TEXT) ENCRYPTION='Y';
+
+INSERT INTO t2 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t2 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t2 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t2 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t2 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t2 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+
+CREATE TABLE t3(a TEXT) ENCRYPTION='Y';
+
+INSERT INTO t3 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t3 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t3 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t3 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t3 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+INSERT INTO t3 (a) VALUES ('Abracadabra is of unknown origin, and its first occurrence is');
+
+SET GLOBAL innodb_page_cleaner_disabled_debug=OFF;
+SET GLOBAL innodb_buf_flush_list_now=ON;
+SET GLOBAL innodb_fast_shutdown=2;
+--source include/shutdown_mysqld.inc
+
+--echo # Writes to Parallel dblwr are not encrypted, so it should be "Pattern found"
+--let $grep_pattern= first occurrence
+--let $grep_file= $MYSQLD_DATADIR/xb_doublewrite
+--let $grep_output= boolean
+--source include/grep_pattern.inc
+
+--source include/start_mysqld.inc
+DROP TABLE t1, t2, t3;
+CREATE TABLE t1(a TEXT) ENCRYPTION='Y';
```

Comment:
What does the 2nd half of the testcase test?

Reply:
In first half, doublewrite buffer is not encrypted, so the contents are visible in xb_doublewrite file "Pattern found" case.

In second half, doublewrite buffer encryption is turned on, the contents shouldn't be readable in xb_doublewrite file. "Pattern not found" case.